src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordClone.cs

```diff
@@ -129,5 +136,28 @@ internal override void GenerateMethodBody(TypeCompilationState compilationState,
 
             throw ExceptionUtilities.Unreachable;
         }
+
+        internal static MethodSymbol? FindValidCloneMethod(NamedTypeSymbol containingType)
+        {
+            for (; !(containingType is null); containingType = containingType.BaseTypeNoUseSiteDiagnostics)
+            {
+                foreach (var member in containingType.GetMembers(WellKnownMemberNames.CloneMethodName))
+                {
+                    if (member is MethodSymbol
+                    {
+                        DeclaredAccessibility: Accessibility.Public,
+                        IsStatic: false,
+                        IsAbstract: false,
+                        IsVirtual: true,
+                        ParameterCount: 0,
+                        Arity: 0
+                    } method)
+                    {
+                        return method;
```

Comment:
>return method; [](start = 24, length = 14)

Post preview we should probably deal with ambiguity cases. Two or more methods match